modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -2382,12 +2403,14 @@ src.size(), fx, and fy; the type of dst is the same as of src.
 @param fy scale factor along the vertical axis; when it equals 0, it is computed as
 \f[\texttt{(double)dsize.height/src.rows}\f]
 @param interpolation interpolation method, see #InterpolationFlags
+@param coordinate coordinate transformation method, see #CoordinateTransformationFlags
 
 @sa  warpAffine, warpPerspective, remap
  */
 CV_EXPORTS_W void resize( InputArray src, OutputArray dst,
                           Size dsize, double fx = 0, double fy = 0,
-                          int interpolation = INTER_LINEAR );
+                          int interpolation = INTER_LINEAR,
+                          int coordinate = INTER_HALF_PIXEL );
```

Comment:
I'm now working on a new function like
``` cpp
void cv::resizeONNX(
    InputArray src, OutputArray dst, Size dsize, Point2d scale = Point2d(),
    unsigned interpolation = INTER_LINEAR | INTER_HALF_PIXEL | INTER_ANTIALIAS,
    double cubicCoeff = -0.75, Rect2d roi = Rect2d())
```
leaving cv::resize unchanged. 

I want to package arguments of coordinate or pixels in one `unsigned`, such as the sampler (4 bit, nearest linear cubic) / coordiante (3bit, half pixel ...) / nearest mode (3 bit, ceil floor prefer_ceil...) antialias (1 bit), to reduce the number function parameters. The extra bits are preserved for future extensions (if have). Maybe we need other ideas?

---

> Modes are usually used with fixed values. So no need to define them as parameters. We could specify modes in function names.

ONNX (Resize), Pytorch (nn.function.interpolate), TensorFlow (image.resize_images), pillow (Image.resize), present OpenCV (cv::resize), all do resize in one function. If we want do some test about different sampler (say for our CNN training), we just change a argument `interpolation` outside, no need to change the function call. Otherwise we must write something tedious like

``` cpp
int sampler       = cfg["sampler"   ];
int antialias     = cfg["antialias" ];
int coordiante    = cfg["coordiate" ];
int nnmode        = cfg["nearest"   ];
double cubicCoeff = cfg["cubicCoeff"];
     if (sampler == INTER_NEAREST) resizeNN    (..., nnmode);
else if (sampler == INTRE_LINEAR ) resizeLinear(..., antialias);
else if (sampler == INTER_CUBIC  ) resizeCubic (..., antialias, cubicCoeff);
```

vs 

``` cpp
int sampler       = cfg["sampler"   ];
int antialias     = cfg["antialias" ];
int coordiante    = cfg["coordiate" ];
int nnmode        = cfg["nearest"   ];
double cubicCoeff = cfg["cubicCoeff"];
resizeONNX(src, dst, dsize, scale, sampler | coordiante | antialias | nnmode, cubicCoeff);
```

And, no matter what sampler we choose, there are many same codes before we do real work with `paralle_for(...Invorker).`